### PR TITLE
Remove all request snapshot code

### LINKED ISSUE #10526: Remove all request snapshot code
### Is your feature request related to a problem? Please describe:
Request snapshot is used to request a snapshot explicitly from follower to fix problems like corruption. But the API has never been used or tested. Technically, it's an unreleased feature. I propose to remove all related code for now to make code base clean.

### PR DESCRIPTION
**Pull request by jiangyuzhao**
### What problem does this PR solve?

Issue Number: close #10526

Problem Summary: 

Request snapshot is used to request a snapshot explicitly from follower to fix problems like corruption. But the API has never been used or tested. Technically, it's an unreleased feature. So we can remove these code to make code clean.

### What is changed and how it works?

Just remove the code about request snapshot

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

File: components/error_code/src/raft.rs [hunk 1]
```diff
@@ -31,8 +30,8 @@ impl ErrorCodeExt for Error {
             Error::CodecError(_) => CODEC_ERROR,
             Error::Exists { .. } => EXISTS,
             Error::NotExists { .. } => NOT_EXISTS,
-            Error::RequestSnapshotDropped => REQUEST_SNAPSHOT_DROPPED,
             Error::ConfChangeError(_) => CONF_CHANGE_ERROR,
+            _ => unreachable!(),
```
Comment (BusyJay) [inline components/error_code/src/raft.rs:34]: I suggest to write the type explicitly to avoid accidentally missing new definitions in the future.

**Review by BusyJay — APPROVED** [review]
/cc @overvenus @NingLin-P

**Review by NingLin-P — APPROVED** [review]
LGTM

**Pull request merged**